src/zeta/aca_oam_server.cpp

```diff
@@ -0,0 +1,323 @@
+                 [](unsigned char c) { return std::tolower(c); });
+  std::replace(mac_string.begin(), mac_string.end(), '-', ':');
+}
+
+string ACA_Oam_Server::_get_vpc_id(uint8_t *vni)
+{
+  string vpc_id;
+  stringstream ss;
+
+  for (int i = 0; i < 3; i++) {
+    ss << std::hex << std::setw(2) << std::setfill('0')
+       << static_cast<unsigned int>(vni[i]);
+  }
+
+  ss >> vpc_id;
+  vpc_id.pop_back();
+
+  return vpc_id;
+}
+
+//extract data for flow table matching from the oam message
+oam_match ACA_Oam_Server::_get_oam_match_field(oam_message *oammsg)
+{
+  oam_match match;
+
+  flow_inject_msg msg_data = oammsg->data.msg_inject_flow;
+
+  match.sip = inet_ntoa(msg_data.inner_src_ip);
+  match.dip = inet_ntoa(msg_data.inner_dst_ip);
+  match.sport = to_string(ntohs(msg_data.src_port));
+  match.dport = to_string(ntohs(msg_data.dst_port));
+  match.proto = to_string(msg_data.proto);
+  match.vni = _get_vpc_id(msg_data.vni);
+
+  return match;
+}
+
+//extract the data used for the flow table action from the oam message
+oam_action ACA_Oam_Server::_get_oam_action_field(oam_message *oammsg)
+{
+  oam_action action;
+
+  flow_inject_msg msg_data = oammsg->data.msg_inject_flow;
+
+  action.inst_nw_dst = inet_ntoa(msg_data.inst_dst_ip);
+  action.node_nw_dst = inet_ntoa(msg_data.node_dst_ip);
+  action.inst_dl_dst = _get_mac_addr(msg_data.inst_dst_mac);
+  _standardize_mac_address(action.inst_dl_dst);
+  action.node_dl_dst = _get_mac_addr(msg_data.node_dst_mac);
+  _standardize_mac_address(action.node_dl_dst);
+  action.idle_timeout = to_string(msg_data.idle_timeout);
+
+  return action;
+}
+
+//check whether the udp_dport is the oam server port of the vpc
+bool ACA_Oam_Server::_check_oam_server_port(uint32_t udp_dport, oam_match match)
+{
+  uint32_t oam_port_of_vpc;
+  aca_vlan_manager::ACA_Vlan_Manager::get_instance().get_oam_server_port(
+          match.vni, &oam_port_of_vpc);
+
+  if (udp_dport == oam_port_of_vpc) {
+    ACA_LOG_INFO("%s", "oam server port is correct!\n");
+    return true;
+  } else {
+    ACA_LOG_ERROR("%s", "oam server port is incorrect!!!");
+    return false;
+  }
+}
+
+void ACA_Oam_Server::_parse_oam_flow_injection(uint32_t udp_dport, oam_message *oammsg)
+{
+  unsigned long not_care_culminative_time;
+  int overall_rc = EXIT_SUCCESS;
+
+  oam_match match = _get_oam_match_field(oammsg);
+
+  // check whether the udp_dport is the oam server port of the vpc
+  if (!_check_oam_server_port(udp_dport, match)) {
+    return;
+  }
+
+  oam_action action = _get_oam_action_field(oammsg);
+
+  string remote_host_ip = action.node_nw_dst;
+  string vpc_id = match.vni;
+  alcor::schema::NetworkType network_type = alcor::schema::NetworkType::VXLAN;
+
+  if (!aca_is_port_on_same_host(remote_host_ip)) {
+    ACA_LOG_INFO("%s", "port_neighbor not exist!\n");
+    string neighbor_id;
+    // get netigbor_id
+
+    //crate neighbor_port
+    aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().create_or_update_l2_neighbor(
```

Comment:
I don't think you can reuse the same fuction create_or_update_l2_neighbor. 

For Zeta enabled port, looks like we can still do the outport for both ZGC gateway path and direct path. Let me know if that's not doable. 

That's the source side, for the destination side, we don't expect an OAM packet to go there to setup direct path (until the destination side send packet to the source side). In order for the destination side to receive the packet, we need to create the "outport". The idea is during the creation Zeta supported ports, it should create a generic "outport" to accept all vxlan packets. 

If this generic "outport" for all vxlan packets would have problem working with the current aca remote IP specific "outport", we will then switch all the aca implementation for both Zeta supported and non-zeta supported to use this generic "outport".